src/sympc/tensor/mpc_tensor.py

```diff
@@ -695,6 +754,80 @@ def __repr__(self):
         """
         return self.__str__()
 
+    def __getattribute__(self, attr_name: str) -> Any:
+        """Get the attribute and check if we should track the gradient.
+
+        Args:
+            attr_name (str): The name of the attribute
+
+        Returns:
+            The attribute specific for this instance
+        """
+        # TODO: Fix this
+        from sympc.tensor.grads import GRAD_FUNCS
```

Comment:
Q: why we are importing this in middle ? do we get any error ?

Reply:
because of circular dependency - it has a TODO